homeassistant/components/meater/sensor.py

```diff
@@ -0,0 +1,175 @@
+
+        return devices
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="meater_api",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=30),
+    )
+
+    def null_callback():
+        return
+
+    # Add a subscriber to the coordinator that doesn't actually do anything, just so that it still updates when all probes are switched off
+    coordinator.async_add_listener(null_callback)
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+
+class MeaterProbeTemperature(CoordinatorEntity):
+    """Meater Temperature Sensor Entity."""
+
+    def __init__(self, coordinator, device_id, temperature_reading_type):
+        """Initialise the sensor."""
+        super().__init__(coordinator)
+        self.coordinator = coordinator
+        self.device_id = device_id
+        self.temperature_reading_type = temperature_reading_type
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"Meater Probe {self.temperature_reading_type.name}"
+
+    @property
+    def state(self):
+        """Return the temperature of the probe."""
+        # First find the right probe in the collection
+        device = None
+
+        for dev in self.coordinator.data:
+            if dev.id == self.device_id:
+                device = dev
+
+        if device is None:
+            return None
+
+        if TemperatureMeasurement.Internal == self.temperature_reading_type:
+            return device.internal_temperature
+
+        # Not an internal temperature, must be ambient
+        return device.ambient_temperature
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+
+        # Meater API always return temperature in Celsius
+        return TEMP_CELSIUS
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return DEVICE_CLASS_TEMPERATURE
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, self.device_id)
+            },
+            "name": "Meater Probe",
+            "manufacturer": "Apption Labs",
+        }
+
+    @property
+    def available(self):
+        """Return if entity is available."""
+        if not self.coordinator.last_update_success:
+            return False
+
+        # See if the device was returned from the API. If not, it's offline
+        return any(self.device_id == device.id for device in self.coordinator.data)
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for the sensor."""
+        return f"{self.device_id}-{self.temperature_reading_type}"
+
```

Comment:
This can be set as entity attribute in the constructor

Reply:
Fixed in 143f8b4